server/app/services/program/ProgramService.java

```diff
@@ -223,27 +223,26 @@ public ImmutableList<ProgramDefinition> getAllProgramDefinitionVersions(long pro
   }
 
   private CompletionStage<ProgramDefinition> syncProgramAssociations(ProgramModel program) {
-    if (isActiveOrDraftProgram(program)) {
+    VersionModel activeVersion = versionRepository.getActiveVersion();
+    VersionModel maxVersionForProgram =
+        programRepository.getVersionsForProgram(program).stream()
+            .max(Comparator.comparingLong(p -> p.id))
```

Comment:
**suggestion (non-blocking):** I *think* this could be replaced with `Program::id`. You can decide if that's better or not.

Reply:
Tried but I think because id is a property not a method in this case it didn't work.